net/dnsdist/Makefile

```diff
@@ -0,0 +1,60 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=dnsdist
+PKG_VERSION:=1.3.3
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
+PKG_SOURCE_URL:=https://downloads.powerdns.com/releases/
+PKG_HASH:=9fb24f9032025955169f3c6e9b0a05b6aa9d6441ec47da08d22de1c1aa23e8cf
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
+
+PKG_MAINTAINER:=James Taylor <james@jtaylor.id.au>
+PKG_LICENCE:=GPL-2.0
+PKG_LICENCE_FILES:=COPYING
+
+PKG_BUILD_DEPENDS:=boost
+
+PKG_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/dnsdist
+  SECTION:=base
+  CATEGORY:=Network
+  TITLE:=dnsdist DNS-, DOS- and abuse-aware loadbalancer.
+  DEPENDS:=+protobuf +re2 +libedit +libfstrm +libsodium +lua +boost +libopenssl +libnetsnmp
+  URL:=http://dnsdist.org/
+endef
+
+define Package/dnsdist/description
+dnsdist is a highly DNS-, DoS- and abuse-aware loadbalancer. Its goal in life
+  is to route traffic to the best server, delivering top performance to legitimate
+  users while shunting or blocking abusive traffic.
+endef
+
+CONFIGURE_ARGS+= \
+	--prefix=/usr \
+	--sysconfdir=/etc \
+	--enable-dnscrypt \
+	--enable-dns-over-tls \
+	--enable-fstrm \
+	--enable-libsodium \
+	--enable-protobuf \
+	--enable-re2 \
+	--with-net-snmp \
+	--enable-libssl
```

Comment:
This should preferably be optional between OpenSSL and GnuTLS

Reply:
I've added the ability to switch between openssl, gnutls or no ssl and added a catch on the dns-over-tls to ensure it's only enabled when ssl is present.